Rubberduck.Parsing/VBA/DeclarationResolveRunner.cs

```diff
@@ -0,0 +1,62 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Rubberduck.VBEditor;
+
+namespace Rubberduck.Parsing.VBA
+{
+    public class DeclarationResolveRunner : DeclarationResolveRunnerBase
+    {
+        private const int _maxDegreeOfDeclarationResolverParallelism = -1;
+
+        public DeclarationResolveRunner(
+            RubberduckParserState state, 
+            IParserStateManager parserStateManager, 
+            ICOMReferenceManager comReferenceManager) 
+        :base(
+             state, 
+             parserStateManager, 
+             comReferenceManager)
+        { }
+
+        public override void ResolveDeclarations(ICollection<QualifiedModuleName> modules, CancellationToken token)
+        {
+            if (!modules.Any())
+            {
+                return;
+            }
+
+            _projectDeclarations.Clear();
+
+                token.ThrowIfCancellationRequested();
```

Comment:
The exceptions are actually used to get directly to the designated point where we go whenever the parsing run is cancelled. Returning instead would just mean that we throw one level further up in the call stack.